libs/domain/product/src/mixins/product.mixin.ts

```diff
@@ -34,15 +30,25 @@ export const ProductMixin = <
   class ProductMixinClass extends superClass {
     @signalProperty({ reflect: true }) sku?: string;
 
-    protected productController = new ProductController(this);
+    protected productService = resolve(ProductService, null);
+
+    protected contextController = new ContextController(this);
 
-    @asyncState()
-    protected product = valueType(this.productController.getProduct());
+    protected $productContext = signal(
+      this.contextController.get<string>(ProductContext.SKU)
+    );
 
-    protected $product = signal(this.productController.getProduct(), {
-      initialValue: null,
+    protected $product = computed(() => {
+      const sku = this.$productContext() || this.sku;
+      return sku ? this.productService?.get({ sku }) : of(null);
     });
+
+    @elementEffect()
+    protected setProductContext = () => {
+      if (this.sku && this.$productContext() !== this.sku) {
+        this.contextController.provide(ProductContext.SKU, this.sku);
+      }
+    };
```

Comment:
is it correct that we should provide context sku inside mixin?
In this case we are going to pollute dom

As far I know we need it only for card component, don't we?